### PR TITLE
fix: fix slow favoriting bug

### PR DESCRIPTION
**Pull request by kayleegeorge**
fix favoriting bug by copying the array by value (not by address). Favorite and unfavorite a token should be ~speedy ~ (and update some color scheme stuff)

Screen Shot 2022-07-05 at 1 20 16 PM

### CODE REVIEW

File: src/components/Explore/TokenRow.tsx [hunk 1]
```diff
@@ -462,35 +462,32 @@ export default function LoadedRow({
...
   )
-  const theme = useTheme()
 
   /* handle favorite token logic */
-  const isTokenFavorited = atom<boolean>(favoriteTokens.includes(tokenAddress))
-  const [tokenFavorited, setTokenFavorite] = useAtom(isTokenFavorited)
   const toggleFavoriteToken = () => {
-    let updatedFavoriteTokens = favoriteTokens
-    if (tokenFavorited) {
+    let updatedFavoriteTokens = favoriteTokens.slice()
```
Comment (vm) [inline src/components/Explore/TokenRow.tsx:482]: `[...favoritedTokens]`

File: src/components/Explore/TokenRow.tsx [hunk 2]
```diff
@@ -502,8 +499,8 @@ export default function LoadedRow({
         <ClickFavorited onClick={() => toggleFavoriteToken()}>
           <Heart
             size={15}
-            color={tokenFavorited ? theme.primary1 : undefined}
-            fill={tokenFavorited ? theme.primary1 : undefined}
+            color={favoriteTokens.includes(tokenAddress) ? theme.primary1 : undefined}
```
Comment (vm) [inline src/components/Explore/TokenRow.tsx:502]: make a const for `favorited`?

**Review by vm — CHANGES_REQUESTED** [review]
couple comments but looking good

File: src/components/Explore/TokenRow.tsx [hunk 3]
```diff
@@ -462,35 +462,32 @@ export default function LoadedRow({
...
   const toggleFavoriteToken = () => {
-    let updatedFavoriteTokens = favoriteTokens
-    if (tokenFavorited) {
+    let updatedFavoriteTokens = favoriteTokens.slice()
+    if (favoriteTokens.includes(tokenAddress)) {
       updatedFavoriteTokens = favoriteTokens.filter((address: string) => {
         return address !== tokenAddress
       })
     } else {
       updatedFavoriteTokens.push(tokenAddress)
```
Comment (vm) [inline src/components/Explore/TokenRow.tsx:488]: then make this `updatedFavoriteTokens = [...favoriteTokens, tokenAddress]`

File: src/components/Explore/TokenRow.tsx (hunk 1 above)
Comment (kayleegeorge) [reply src/components/Explore/TokenRow.tsx:482]: Screen Shot 2022-07-05 at 2 19 19 PM

Comment (kayleegeorge) [reply src/components/Explore/TokenRow.tsx:482]: for some reason this is the same bug where it doesn't re-render

File: src/components/Explore/TokenRow.tsx (hunk 3 above)
Comment (kayleegeorge) [reply src/components/Explore/TokenRow.tsx:488]: ok thank you thank you

**Review by vm — APPROVED** [review]

File: src/components/Explore/TokenRow.tsx [hunk 4]
```diff
@@ -462,35 +462,33 @@ export default function LoadedRow({
...
-  const theme = useTheme()
 
   /* handle favorite token logic */
-  const isTokenFavorited = atom<boolean>(favoriteTokens.includes(tokenAddress))
-  const [tokenFavorited, setTokenFavorite] = useAtom(isTokenFavorited)
   const toggleFavoriteToken = () => {
-    let updatedFavoriteTokens = favoriteTokens
-    if (tokenFavorited) {
+    let updatedFavoriteTokens
+    if (favoriteTokens.includes(tokenAddress)) {
```
Comment (vm) [inline src/components/Explore/TokenRow.tsx:484]: `isFavorited`

**Pull request merged**